src/buffers/RecordingBuffer.h

```diff
@@ -76,5 +72,10 @@ namespace timeshift {
     
     void SetDuration(int duration) { m_Duration = duration; XBMC->Log(LOG_ERROR, "XXXXX Duration set to %d XXXXX", m_Duration); }
     int GetDuration(void) { return m_Duration; XBMC->Log(LOG_ERROR, "XXXXX Duration set to %d XXXXX", m_Duration); }
+
+    bool Open(const std::string inputUrl,const PVR_RECORDING &recording);
+    int Read(byte *buffer, size_t length);
```

Comment:
Using 'override' signals the compiler that the function should overload a virtual function in the base, and will generate a warning if it doesn't.   I'd recommend removing it from prototypes unless there's a compelling reason.

Reply:
Open needs to be new because of taking two variables.

Read needs an override and I changed it to match TimeShiftBuffer.h